### PR TITLE
feat: Add `ParseUser::logInAs` method

### PR DESCRIPTION
**Pull request by isaacaskew**
### New Pull Request Checklist

Closes: #485

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-php-sdk/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-php-sdk/issues/485).

### Issue Description
Later versions of Parse Server [allow for the `loginAs` method](https://github.com/parse-community/parse-server/pull/7406). Given a `userId` and the master key, one can become logged in as the user associated to the `userId` provided. This PR adds the PHP Parse SDK method to call the Parse Server endpoint.

### Approach
To implement this method, I've added a function similar to `login` but modified to only require the `userId`, as the master key is already assumed when using this method. Tests similar to the `login` tests have been added and modified to cover the cases of a success or a failure with no `userId` provided.

### CODE REVIEW

**Comment by mtrezza** [comment]
@dplewis Is this something you'd want to review?

**Review by mtrezza — APPROVED** [review]
Looks good

File: src/Parse/ParseUser.php [hunk 1]
```diff
@@ -157,6 +157,33 @@ public static function logIn($username, $password)
...
     }
 
+    /**
+     * Uses the master key to log in and return a valid ParseUser, or throws if invalid.
+     *
+     * @param $userId
+     *
+     * @throws ParseException
+     *
+     * @return static
```
Comment (dplewis) [inline src/Parse/ParseUser.php:167]: `@return ParseUser` to improve documentation

**Review by dplewis — CHANGES_REQUESTED** [review]
I didn't know this was a feature. Probably should add it to the JS SDK

File: src/Parse/ParseUser.php (hunk 1 above)
Comment (isaacaskew) [reply src/Parse/ParseUser.php]: Done

**Review by dplewis — APPROVED** [review]
Thank you for your contribution!

**Comment by mtrezza** [comment]
@dplewis Before we merge this, do you have any idea how the test results are reported here? This hasn't migrated to GitHub Actions yet and is still using Travis CI.

**Comment by dplewis** [comment]
MIght have to reenable travis and pay the bill. I'm currenly working on migrating this to Github Action

https://github.com/dplewis/parse-php-sdk/pull/1

**Comment by isaacaskew** [comment]
@mtrezza , @dplewis I forked this repo and tested it against these Parse Server Versions:

3.10.0 - [[CircleCI run](https://app.circleci.com/pipelines/github/isaacaskew/parse-php-sdk/19/workflows/2f67f69a-c179-4a9e-b5f0-8e7a1cf1c253/jobs/19)] [[GitHub Changes](https://github.com/isaacaskew/parse-php-sdk/commit/ae8bda09c00a5a9df13a3a415af96a1ed001a137)] - `Tests: 598, Assertions: 1334, Errors: 22, Failures: 4, Skipped: 1`

5.4.3 - [[CircleCI run](https://app.circleci.com/pipelines/github/isaacaskew/parse-php-sdk/21/workflows/b2a8bf29-4b9d-4395-8e9c-d905d5a814d7/jobs/21)] [[GitHub Changes](https://github.com/isaacaskew/parse-php-sdk/commit/f5a8ad1702a2610fb4eeff2d986a54e0c19fc90d)] - `Tests: 598, Assertions: 1292, Errors: 29, Failures: 5, Skipped: 1`

6.0.0 - [[CircleCI run](https://app.circleci.com/pipelines/github/isaacaskew/parse-php-sdk/22/workflows/bb36d721-51b7-4510-a0ea-0b1695140357/jobs/22)] [[GitHub Changes](https://github.com/isaacaskew/parse-php-sdk/commit/67e7d1a153b804579b27a32536859f6b9a88ab76)] - `Tests: 598, Assertions: 1272, Errors: 36, Failures: 5, Skipped: 1`

```
There were 5 failures:

1) Parse\Test\ParseCloudTest::testGetJobsData
Failed asserting that 0 matches expected 3.

/Users/isaac/Development/parse-php-sdk/tests/Parse/ParseCloudTest.php:124

2) Parse\Test\ParseSessionFixationTest::testCookieIdChangedForAnonymous
Failed asserting that '' is not equal to "".

/Users/isaac/Development/parse-php-sdk/tests/Parse/ParseSessionFixationTest.php:51

3) Parse\Test\ParseSessionFixationTest::testCookieIdChangedForAnonymousToRegistered
Failed asserting that '' is not equal to "".

/Users/isaac/Development/parse-php-sdk/tests/Parse/ParseSessionFixationTest.php:65

4) Parse\Test\ParseSessionStorageTest::testIsUsingParseSession
Failed asserting that false is true.

/Users/isaac/Development/parse-php-sdk/tests/Parse/ParseSessionStorageTest.php:48

5) Parse\Test\ParseUserTest::testCountUsers
Failed asserting that 1 matches expected 3.

/Users/isaac/Development/parse-php-sdk/tests/Parse/ParseUserTest.php:587
```

Note that there were only 4 error when run against Parse 3, and 5 errors in the other two. Let me know if there's anything about the server in particular I need to configure for these other ones to pass - otherwise I can try and help fix these once I understand a bit more of the context around a few of these functions.

In the meantime, [Parse 6 running with my tests passes](https://app.circleci.com/pipelines/github/isaacaskew/parse-php-sdk/23/workflows/609aecc4-4882-4848-91f4-f10ddf8d98ec/jobs/26) - we can see here that the pipeline run [with my changes](https://github.com/isaacaskew/parse-php-sdk/commit/4abfd69abd06116c8c45ac336ad0b2289f5b6e76) has the same number of errors as before, the 5 above. I chose Parse 6 as the server to run the tests against since Parse 6 is the version that introduced the new endpoint that I needed the client to talk to. I'm not in a rush to merge this PR so I'm game to help get the other tests passing and everything running clean against every version before we merge my feature in.

I can target the changes against a new major version of Parse PHP SDK as well - that might make sense since this method wouldn't work for any folks who upgrade their PHP client after this change is released but are using Parse Server versions < 6.

**Comment by mtrezza** [comment]
We've released the new SDK version 2.0.0. I've rebased this PR, let's see how tests are doing.

To your point, I think we should add a compatibility table to the README for Parse Server, to indicate which Parse Server versions this SDK is compatible with.

**Comment by mtrezza** [comment]
I suppose we can merge this, it seems that #491 contains the same tests as this PR?

**Comment by dplewis** [comment]
@mtrezza I think this is good to merge. You want to try auto release with this one?

**Comment by mtrezza** [comment]
Yes, let's give it a try!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [2.1.0](https://github.com/parse-community/parse-php-sdk/releases/tag/2.1.0)

**Pull request merged**